Allow filtering documents with tag synonyms

fugal-dy opened the pull request:
This adds the TagSynonymGroup model and extends the
documents tagfilter such that filtering for document
by tag looks for associated synonyms to include.

Issue #133

winged reviewed (changes requested):
Good work so far!

There are a couple of things missing:

* Register the new viewset under **alexandria/core/urls.py**
* Your `black` was a bit overzealous - check the version we use vs. your version

fugal-dy commented on alexandria/core/models.py:
```diff
@@ -8,6 +8,7 @@
 from django.utils.translation import gettext_lazy as _
 from localized_fields.fields import LocalizedCharField, LocalizedTextField
 from rest_framework import exceptions
+from django.conf import settings
```
```suggestion
```
True. Incomplete code cleanup after removing localization that was using `settings.LANGUAGES` for `choices`  and `settings.LANGUAGE_CODE` for `default`

fugal-dy commented:
I have applied your inputs, fixed the filter (did not do what it should) and did some cleanup. On top I added an extra test that does not make it past linting though o_0. I need some sleep now, will get back to finishing this later

Yelinz commented:
Seems to have been merged with https://github.com/projectcaluma/alexandria/pull/256 already, reopen if not
